### PR TITLE
More constants for kernel ASM

### PR DESCRIPTION
**Pull request by dlubarov**
- `GlobalMetadata` - offsets for global kernel variables in memory
- `ContextMetadata` - offsets for context-specific kernel variables in memory
- `GAS_CONSTANTS`, based on the yellowpaper

Also move constants to a separate module since `aggregator` was getting long.

Depends on #650.

### CODE REVIEW

File: evm/src/cpu/kernel/global_metadata.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/// These metadata fields contain global VM state, stored in the `Segment::Metadata` segment of the
+/// kernel's context (which is zero).
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Ord, PartialOrd, Debug)]
+pub(crate) enum GlobalMetadata {
+    /// The larger context ID that has been used so far in this execution. Tracking this allows us
```
Comment (nbgl) [inline evm/src/cpu/kernel/global_metadata.rs:5]:
```suggestion
    /// The largest context ID that has been used so far in this execution. Tracking this allows us
```

**Review by nbgl — APPROVED** [review]

File: evm/src/cpu/kernel/context_metadata.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/// These metadata fields contain VM state specific to a particular context.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Ord, PartialOrd, Debug)]
+pub(crate) enum ContextMetadata {
+    /// The ID of the context which created this one.
+    ParentContext = 0,
+    /// The program counter to return to when we return to the parent context.
+    ParentProgramCounter = 1,
+    /// If we're in a system call, this holds the userspace program counter to return to.
+    UserspaceProgramCounter = 2,
```
Comment (nbgl) [inline evm/src/cpu/kernel/context_metadata.rs:9]: Oh, in #653 I put put that on the stack

Comment (nbgl) [reply evm/src/cpu/kernel/context_metadata.rs:9]: This also permits nested syscalls/traps.

Comment (dlubarov) [reply evm/src/cpu/kernel/context_metadata.rs:9]: Ah right, forgot we decided that. Removed this.

**Pull request merged**